tools/amGithubNotifier/main.go

```diff
@@ -0,0 +1,187 @@
+	defaultRepo  string
+	portNo       string
+	dryRun       bool
+}
+
+type ghWebhookReceiver struct {
+	ghClient *github.Client
+	cfg      ghWebhookReceiverConfig
+}
+
+type ghWebhookHandler struct {
+	client *ghWebhookReceiver
+}
+
+func main() {
+	/*
+		Example `alerts.rules.yml`:
+		```
+		groups:
+		- name: groupname
+		  rules:
+		  - alert: alertname
+		    expr: up == 0
+		    for: 2m
+		    labels:
+		      severity: average
+		      prNum: '{{ $labels.prNum }}'
+		      owner: prometheus (optional)
+		      repo: prometheus (optional)
+		    annotations:
+		      description: 'description of the alert'
+		```
+	*/
+	cfg := ghWebhookReceiverConfig{}
+
+	app := kingpin.New(filepath.Base(os.Args[0]), `alertmanager github webhook receiver
+	Example: ./amGithubNotifier --org=prometheus --repo=prometheus --port=8080`)
+	app.Flag("authfile", "path to github oauth token file").Default("/etc/github/oauth").StringVar(&cfg.authFile)
+	app.Flag("org", "default org/owner").Required().StringVar(&cfg.defaultOwner)
+	app.Flag("repo", "default repo").Required().StringVar(&cfg.defaultRepo)
+	app.Flag("port", "port number to run the server in").Default("8080").StringVar(&cfg.portNo)
+	app.Flag("dryrun", "dry run for github api").BoolVar(&cfg.dryRun)
+
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	client, err := newGhWebhookReceiver(cfg)
+	if err != nil {
+		log.Fatalf("failed to create GitHub Webhook Receiver client: %v", err)
+	}
+
+	serveWebhook(client)
+}
+
+func (hl ghWebhookHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		log.Printf("unsupported request method: %v: %v", r.Method, r.RemoteAddr)
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	msg := &webhook.Message{}
+	ctx := r.Context()
+
+	// Decode the webhook request.
+	err := json.NewDecoder(r.Body).Decode(msg)
+	if err != nil {
+		log.Println("failed to decode webhook data")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Handle the webhook message.
+	log.Printf("handling alert: %v", alertID(msg))
+	if _, err := hl.client.processAlerts(ctx, msg); err != nil {
+		log.Printf("failed to handle alert: %v: %v", alertID(msg), err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	log.Printf("completed alert: %v", alertID(msg))
+	w.WriteHeader(http.StatusOK)
+}
+
+func newGhWebhookReceiver(cfg ghWebhookReceiverConfig) (*ghWebhookReceiver, error) {
+
+	if cfg.dryRun {
+		return &ghWebhookReceiver{
+			ghClient: github.NewClient(nil),
+			cfg:      cfg,
+		}, nil
+	}
+
+	// add github token
+	oauth2token, err := ioutil.ReadFile(cfg.authFile)
+	if err != nil {
+		return nil, err
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: string(oauth2token)},
+	)
+	ctx := context.Background()
+	tc := oauth2.NewClient(ctx, ts)
+
+	return &ghWebhookReceiver{
+		ghClient: github.NewClient(tc),
+		cfg:      cfg,
+	}, nil
+}
+
+// processAlert formats and posts the alert to github
```

Comment:
```suggestion
// processAlert formats and posts the alert to Github.
```

please check all comments everythere.